nuxeo-quota-core/src/main/java/org/nuxeo/ecm/quota/count/DocumentsCountUpdater.java

```diff
@@ -131,39 +131,24 @@ protected void updateCountStatistics(CoreSession session, DocumentModel doc, Lis
 
         if (!doc.hasFacet(FOLDERISH)) {
             DocumentModel parent = ancestors.get(0);
-            updateParentChildrenCount(session, parent, count);
-        }
-
-        for (DocumentModel ancestor : ancestors) {
-            Number previous;
-            if (ancestor.hasFacet(DOCUMENTS_COUNT_STATISTICS_FACET)) {
-                previous = (Number) ancestor.getPropertyValue(DOCUMENTS_COUNT_STATISTICS_DESCENDANTS_COUNT_PROPERTY);
-            } else {
-                ancestor.addFacet(DOCUMENTS_COUNT_STATISTICS_FACET);
-                previous = null;
-            }
-            DeltaLong descendantsCount = DeltaLong.valueOf(previous, count);
-            ancestor.setPropertyValue(DOCUMENTS_COUNT_STATISTICS_DESCENDANTS_COUNT_PROPERTY, descendantsCount);
-            // do not send notifications
-            QuotaUtils.disableListeners(ancestor);
-            DocumentModel origAncestor = ancestor;
-            session.saveDocument(ancestor);
-            QuotaUtils.clearContextData(origAncestor);
+            updateCount(session, parent, DOCUMENTS_COUNT_STATISTICS_CHILDREN_COUNT_PROPERTY, count);
         }
 
+        ancestors.forEach(ancestor -> updateCount(session, ancestor,
+                DOCUMENTS_COUNT_STATISTICS_DESCENDANTS_COUNT_PROPERTY, count));
```

Comment:
Interesting factoring but if it's unrelated to the main goal of this ticket (trash/untrash) it should be in a separate commit, to make it clear to the reader that there's no trash/untrash logic in it.